src/OPFS/OPFS.worker.js

```diff
@@ -584,62 +579,61 @@ async function downloadModel(objectUrl, shaHash, originalFilePath, owner, repo,
             self.postMessage({completed: true, event: 'renamed', file: updatedBlobFile})
           }
         }
-      }
-    } catch (error) {
-      // expected if file not found - lets see if we have a temporary file
+      } else {
```

Comment:
Nick, this was the fallthru we fixed together.  pls double-check i got the edit right